### PR TITLE
Fix FindOperatorPairs did not find expected number of op_ methods

### PR DESCRIPTION
**Pull request by chamons**
- https://github.com/mono/Embeddinator-4000/issues/207
- Multiple operators of the same type but different arguments would fail unnecessarily
- Move overloads tests to new file due to length.

### CODE REVIEW

**Comment by chamons** [comment]
The diff isn't bad, but someone wants me to do the test move/rename in a different PR I can.

File: objcgen/operatoroverloads.cs [hunk 1]
```diff
@@ -68,13 +68,15 @@ static OperatorOverloads ()
...
-					yield return match.Value[1];
-				else
-					yield return match.Value[0];
+				// If we have a friendly method, ignore op variants and vice versa
+				MethodInfo friendlyMethod = match.Value.FirstOrDefault (x => !x.Name.StartsWith ("op_", StringComparison.InvariantCulture));
+				if (friendlyMethod != null) {
+					foreach (var opMethod in match.Value.Where (x => x.Name.StartsWith ("op_", StringComparison.InvariantCulture)))
+						yield return opMethod;
+				}
+				else {
```
Comment (spouliot) [inline objcgen/operatoroverloads.cs]: minor/style: `} else {`

File: tests/objc-cli/libmanaged/Tests/Tests.m [hunk 2]
```diff
@@ -693,60 +693,63 @@ - (void)testProtocols {
...
-    XCTAssertTrue ([[Methods_AllOperatorsWithFriendly leftShiftC1:oneFriend a:2] value] == (1 << 2), "1 << 2 All");
-    XCTAssertTrue ([[Methods_AllOperatorsWithFriendly rightShiftC1:oneFriend a:2] value] == (1 >> 2), "1 >> 2 All");
-    
-    XCTAssertTrue ([[Methods_AllOperatorsWithFriendly onesComplementC1:oneFriend] value] == (~1), "!1 All");
-    XCTAssertTrue ([[Methods_AllOperatorsWithFriendly negateC1:oneFriend] value] == (-1), "- 1 All");
-    XCTAssertTrue ([[Methods_AllOperatorsWithFriendly plusC1:oneFriend] value] == (+1), "+ 1 All");
-    
-    XCTAssertTrue ([[Methods_AllOperatorsWithFriendly decrementC1:oneFriend] value] == (1 - 1), "1 - 1 All");
-    XCTAssertTrue ([[Methods_AllOperatorsWithFriendly incrementC1:oneFriend] value] == (1 + 1), "1 + 1 All");    
+    Overloads_OperatorCollision * one = [[Overloads_OperatorCollision alloc] initWithVal:1];
```
Comment (spouliot) [inline tests/objc-cli/libmanaged/Tests/Tests.m]: without going back (in this PR) to existing tests please make sure all new ones (and updates) are using tabs (not spaces)

File: objcgen/operatoroverloads.cs [hunk 3]
```diff
@@ -68,13 +68,15 @@ static OperatorOverloads ()
...
 
-				if (match.Value.Count (x => x.Name.StartsWith ("op_", StringComparison.InvariantCulture)) != 1)
-					throw new EmbeddinatorException (99, $"Internal error `FindOperatorPairs did not find expected number of op_ methods`. Please file a bug report with a test case (https://github.com/mono/Embeddinator-4000/issues");
-
-				if (!match.Value[0].Name.StartsWith ("op_", StringComparison.InvariantCulture))
-					yield return match.Value[1];
-				else
-					yield return match.Value[0];
+				// If we have a friendly method, ignore op variants and vice versa
+				MethodInfo friendlyMethod = match.Value.FirstOrDefault (x => !x.Name.StartsWith ("op_", StringComparison.InvariantCulture));
```
Comment (spouliot) [inline objcgen/operatoroverloads.cs]: missed it before, use `StringComparison.Ordinal`

File: objcgen/operatoroverloads.cs [hunk 4]
```diff
@@ -68,13 +68,15 @@ static OperatorOverloads ()
...
+				if (friendlyMethod != null) {
+					foreach (var opMethod in match.Value.Where (x => x.Name.StartsWith ("op_", StringComparison.InvariantCulture)))
```
Comment (spouliot) [inline objcgen/operatoroverloads.cs]: same

**Pull request merged**